### PR TITLE
Factor out integration test scenarios for readability

### PR DESCRIPTION
**Pull request by KtorZ**
<p align="right">#172</p>

# Overview

- [ ] I have grouped test scenarios related to a single requirement together 
- [ ] I have factored out similar scenarios to reduce verbosity (introducing test matrices) 

# Comments

Also, in the meantime, reviewed indentation and align scenarios names
with the requirements matrix.
@piotr-iohk Lot of things were indented with 2-spaces whereas we do use 4 in the project. I believe this was before you set up the `.editorconfig` and it should be fine now :+1:

### CODE REVIEW

**Comment by KtorZ** [comment]
The diff isn't much pretty in `Scenarios/Wallets` since I had to re-indent everything :s

**Comment by KtorZ** [comment]
Note that the output for the wallet test now looks like this:

```
Wallets
  WALLETS_DELETE_01 - deleted wallet is not available
  WALLETS_LIST_01 - One can list all wallets without providing any parameters
  WALLETS_UPDATE_01 - updating a wallet persists the update
  WALLETS_UTXO_03 - UTxO statistics reflect wallet's inactivity
  WALLETS_UTXO_04 - UTxO statistics reflect wallet's activity
  WALLETS_CREATE_04 - Cannot perform operation other than 'create' or 'restore'
    Invalid operation name
    Empty operation name
  CreateWallet
    WALLETS_CREATE_01 - One can create/restore previously deleted wallet
    WALLETS_CREATE_02 - One can create/restore wallet without spending password
    WALLETS_CREATE_03 - One can create/restore wallet with spending password
    WALLETS_CREATE_04 - One cannot create/restore wallet if spending password is not hex-encoded string
      spending password is less than 32 bytes / 64 characters
      spending password is more than 32 bytes / 64 characters
      spending password is a number
      spending password is an empty string
      spending password is an array
      spending password is an arbitrary utf-8 string
    WALLETS_CREATE_05 - One cannot create/restore wallet that already exists
    WALLETS_CREATE_07 - Cannot create/restore wallet using less or more than 12 mnemonics which are valid BIP-39
      less than 12 words
      more than 12 words
      empty mnemonic
    WALLETS_CREATE_08 - One cannot create wallet using invalid BIP-39 mnemonics
      Kanji mnemonics / non-English mnemonics
      Invalid mnemonics
    WALLET_CREATE_09 - One cannot create wallet with mnemonics from the API doc
    WALLETS_CREATE_10 - One cannot create wallet with assurance level other than 'strict' or 'normal'
      Arbitrary String
      Empty String
    WALLETS_CREATE_11 - One cannot perform POST to api/v1/wallets without all required parameters (operation)
      Missing operation
      Missing backupPhrase
      Missing assuranceLevel
      Missing name
    WALLETS_CREATE_12 - One can create/restore wallet with name of any length
      long wallet name
      empty name
  RestoreWallet
    WALLETS_CREATE_01 - One can create/restore previously deleted wallet
    WALLETS_CREATE_02 - One can create/restore wallet without spending password
    WALLETS_CREATE_03 - One can create/restore wallet with spending password
    WALLETS_CREATE_04 - One cannot create/restore wallet if spending password is not hex-encoded string
      spending password is less than 32 bytes / 64 characters
      spending password is more than 32 bytes / 64 characters
      spending password is a number
      spending password is an empty string
      spending password is an array
      spending password is an arbitrary utf-8 string
    WALLETS_CREATE_05 - One cannot create/restore wallet that already exists
    WALLETS_CREATE_07 - Cannot create/restore wallet using less or more than 12 mnemonics which are valid BIP-39
      less than 12 words
      more than 12 words
      empty mnemonic
    WALLETS_CREATE_08 - One cannot create wallet using invalid BIP-39 mnemonics
      Kanji mnemonics / non-English mnemonics
      Invalid mnemonics
    WALLET_CREATE_09 - One cannot create wallet with mnemonics from the API doc
    WALLETS_CREATE_10 - One cannot create wallet with assurance level other than 'strict' or 'normal'
      Arbitrary String
      Empty String
    WALLETS_CREATE_11 - One cannot perform POST to api/v1/wallets without all required parameters (operation)
      Missing operation
      Missing backupPhrase
      Missing assuranceLevel
      Missing name
    WALLETS_CREATE_12 - One can create/restore wallet with name of any length
      long wallet name
      empty name
```

File: test/integration/Test/Integration/Scenario/Wallets.hs [hunk 1]
```diff
@@ -830,17 +87,296 @@ spec = do
...
+                [ ( "Invalid operation name"
+                  , [json| #{russianWalletName} |]
+                  , [ expectJSONError "Error in $.operation: When parsing Cardano.Wallet.API.V1.Types.WalletOperation expected a String with the tag of a constructor but got" ]
+                  )
+                , ( "Empty operation name"
+                  , [json| "" |]
+                  , [ expectJSONError "Error in $.operation: When parsing Cardano.Wallet.API.V1.Types.WalletOperation expected a String with the tag of a constructor but got" ]
+                  )
+                ]
+        forM_ matrix $ \(title, operation, expectations) -> scenario title $ do
```
Comment (piotr-iohk) [inline test/integration/Test/Integration/Scenario/Wallets.hs]: I assume that if the test fails on the particular `matrix` entry it moves to the next one?

**Review by piotr-iohk — COMMENTED** [review]
That is a really nice cleanup :) 2x less code and a lot more readable now.

File: test/integration/Test/Integration/Scenario/Wallets.hs (hunk 1 above)
Comment (piotr-iohk) [reply test/integration/Test/Integration/Scenario/Wallets.hs]: ok, yes it does... :)

**Comment by KtorZ** [comment]
*edit: fixed a few titles to be 100% aligned with the requirements matrix.

**Review by piotr-iohk — APPROVED** [review]
OK

**Pull request merged**